### PR TITLE
CDSDX-4014 enable configuration as default for SDX;

### PR DESCRIPTION
**Pull request by RokolyA94**
See detailed description in the commit message.

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseRestKnoxServiceConfigProvider.java [hunk 1]
```diff
@@ -29,17 +25,9 @@ public class HbaseRestKnoxServiceConfigProvider implements CmTemplateComponentCo
...
-
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject templatePreparationObject) {
-        List<ApiClusterTemplateConfig> configs = new ArrayList<>();
-        configs.add(config(RESTSERVER_SECURITY_AUTHENTICATION, "kerberos"));
-        if (isSDXOptimizationNeeded(templatePreparationObject)) {
-            configs.add(config(HBASE_RPC_PROTECTION, "privacy"));
-        }
-        return configs;
+        return List.of(config(RESTSERVER_SECURITY_AUTHENTICATION, "kerberos"), config(HBASE_RPC_PROTECTION, "privacy"));
```
Comment (kkalvagadda1) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseRestKnoxServiceConfigProvider.java:30]: This change should still be under an entitlement. we know there are integration DH issues with this rpc protection configuration.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java [hunk 2]
```diff
@@ -92,8 +92,8 @@ protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, Templat
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
         List<ApiClusterTemplateConfig> configs = new ArrayList<>();
         if (isSDXOptimizationEnabled(source)) {
-            configs.add(config(DFS_ENCRYPT_DATA_TRANSFER, "true"));
             configs.add(config(DFS_DATA_TRANSFER_PROTECTION, "privacy"));
+            configs.add(config(DFS_ENCRYPT_DATA_TRANSFER, "true"));
```
Comment (kkalvagadda1) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java:96]: What is the change here?

**Review by romedeiro — APPROVED** [review]

**Review by romedeiro — APPROVED** [review]

**Comment by doktoric** [comment]
@RokolyA94 broken checkstyle

**Review by votez — APPROVED** [review]

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseRestKnoxServiceConfigProvider.java [hunk 3]
```diff
@@ -59,10 +47,4 @@ public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, Te
                 && source.getGatewayView().getExposedServices().contains(exposedServiceCollector.getHBaseRestService().getKnoxService());
     }
 
-    private boolean isSDXOptimizationNeeded(TemplatePreparationObject source) {
-        return entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
-                && source.getStackType() != null
-                && source.getStackType().equals(StackType.DATALAKE);
```
Comment (bergerdenes) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseRestKnoxServiceConfigProvider.java:65]: This check is completely missing.

**Comment by keyki** [comment]
Closing the PR, please re-open if it's still relevant.

**Pull request closed without merging**